R/gateUtil.R

```diff
@@ -106,17 +76,13 @@ commonGateCreate = function(body, name, gid,
 
   body = jsonlite::toJSON(body, null = "null", digits = NA)
   path = paste("experiments", experimentId, "gates", sep = "/")
-  gateResp = basePost(path, body, list())
 
   if (createPopulation) {
-    populationResp = autoCreatePopulation(experimentId, name, gid, parentPopulationId)
-    return(list(
-      gate = gateResp,
-      population = populationResp
-    ))
+      gateResp = basePost(path, body, params=list("createPopulation"="true"))
   } else {
-    return(list(gate = gateResp))
+      gateResp = basePost(path, body, list())
   }
+  return(list(gate = gateResp))
```

Comment:
When the API is called with `?createPopulation=true`, it responds with an object like `{gate: <gate>, population: <population>}` or `{gate: <gate>, populations: <population>[]}`, so you can just return the result of `basePost` in `if (createPopulation)`. (In the `else` case, you still need the `list(gate = gateResp)` to match current behavior.)